io/io/test/TFileMergerTests.cxx

```diff
@@ -219,3 +219,41 @@ TEST(TFileMerger, ChangeFile)
    gSystem->Unlink("file6640mergerinput_2.root");
    gSystem->Unlink("file6640mergeroutput.root");
 }
+
+// https://github.com/root-project/root/issues/9022
+TEST(TFileMerger, SingleHistFile)
+{
+   auto filename1 = "f1_9022.root";
+   auto filename2 = "f2_9022.root";
+   auto outname = "file9022mergeroutput.root";
+   {
+      TFile f1(filename1, "RECREATE");
+      TH1F h("h1", "h1", 1, 0, 1);
+      h.Write();
+      f1.Close();
+      TFile f2(filename2, "RECREATE");
+      TH1F h2("h2", "h2", 1, 0, 1);
+      h2.Write();
+      f2.Close();
+   }
+   {
+      TFileMerger filemerger{false, false};
+      filemerger.SetMaxOpenedFiles(2);
+      filemerger.OutputFile(std::unique_ptr<TFile>{TFile::Open(outname, "RECREATE")});
+
+      filemerger.AddFile(filename1);
+      filemerger.AddFile(filename2);
+
+      // Before the fix, TTree::ChangeFile was called during Merge
```

Comment:
This comment only makes sense in the context of this PR: once we merge it it won't have the context anymore and readers will be left to wonder "which fix?"
Therefore I'd either reference the commit hash or remove the comment

Reply:
That was a typo, sorry, removed now :) thanks for noticing.